README.md
Regelwerk der SKGB
==================

Dieses Repository enthält die **Satzung und Vereinsordnungen** der [SKGB][]
(in `src`). Es soll die Information der Mitglieder verbessern sowie die
Diskussion um die Fortbildung des Regelwerks fördern.


Ich will doch bloß die Satzung lesen!
-------------------------------------

Eine druckfertige PDF-Fassung unserer Satzung mitsamt der Vereinsordnungen
findest Du im Impressum unserer Website (unter [Kontakt][] im Hauptmenü).


Über dieses Repository
----------------------

Der `stable`-Branch soll nur Textfassungen enthalten, welche die offiziellen,
von den Vereinsorganen beschlossenen Regelungen bestmöglich wiedergeben. Er
kommt damit einer verbindlichen Quelle für die Gesamtfassung am nächsten.
Wirklich verbindlich sind indes allein die Wortlaute der jeweiligen Beschlüsse!

Der `master`-Branch ist der Default-Branch. Er dient der Arbeit an
der Umsetzung der Beschlüsse in griffige Formulierungen und korrekte
XML-Auszeichnung. Wenn aktuell keine Änderungen anstehen, sollten `master`
und `stable` somit auf dem gleichen Stand sein.

Weitere Branches können fallweise für Ordnungsentwürfe zur Vorbereitung
von Änderungen oder zu experimentellen Zwecken genutzt werden.

Die Idee, das SKGB-Regelwerk in [XML][] zu kodieren, entstand um 2005 aus der
Notwendigkeit, identische Inhalte sowohl im Web-Browser als auch in gedruckter
Fassung verfügbar zu machen. Damals ließ die Druckqualität aus dem Browser
oft zu wünschen übrig, und PDF- und andere Dokumente konnten meist nur über
Plug-ins im Browser angezeigt werden. [XSL][] versprach die Chance, technisch
neutrale Inhalt je nach Ziel in HTML oder in ein PDF umzuwandeln.

Tatsächlich stellte sich die gewünschte Umwandlung mit XSL wie auch die
notwendige präzise Nachführung der XML-Quelldokumente als eher mühselig
heraus. Mittlerweile existiert jedoch eine Toolchain, welche die Erzeugung
der benötigten Dokumente zumindest soweit vereinfacht, dass nur noch wenig
Handarbeit nötig ist. Die bestehende Technik kann daher bis auf Weiteres
weiter genutzt werden. Perspektivisch könnte theoretisch die bisher gepflegte
Kennzeichnung der Änderungsgeschichte über XML-Attribute entfallen, falls
sich im Vorstand stattdessen die Nutzung dieses Repositories etabliert.

Neue **Issues und Pull Requests** sind in diesem Repository gerne gesehen,
stellen jedoch **keine** Anträge für Regeländerungen dar. Vielmehr erlauben
Issues und Pull Requests hier (neben Fehlerkorrekturen), das Regelwerk offen
zu diskutieren und Ideen zu entwickeln, die später einmal den Vereinsorganen
zur Abstimmung vorgelegt werden könnten oder auch nicht.

Wenn du zeitnah eine konkrete Regeländerung wünschst, nimmst du also am besten
formlos [Kontakt][] mit dem Vereinsvorstand auf, idealerweise mindestens drei
Monate vor der nächsten Mitgliederversammlung.


Abkürzungen
-----------

Die einzelnen Dokumente sind mit Abkürzungen gekennzeichnet, um die Dateinamen
übersichtlich kurz zu halten. Diese Abkürzungen sind größtenteils nicht
allgemein üblich. Sie sollten in der normalen Kommunikation durch die folgenden
umgangssprachlichen Kurznamen ersetzt werden:

**ANutzO** – Allgemeine Vereinsboots-Nutzungsordnung  
**DSE** – Datenschutzerklärung  
**FinO** – Finanzordnung  
**HausO** – Bootshausordnung  
**JO** – Jugendordnung  
**JungSV** – Jüngstensegel-Vorschrift  
**NutzOJuB** – Jugendboot-Nutzungsordnung  
**NutzOVayu** – Sailhorse-Nutzungsordnung  
**NutzOvG** – Lelievlet-Nutzungsordnung  
**StegO** – Steg- und Liegeplatzordnung


Weiterverwendung
----------------

Der Vorstand hat sich bisher mangels Notwendigkeit noch nicht zum Nutzungsrecht
der Inhalte dieses Repositorys durch Dritte positioniert. Es ist jedoch
die persönliche inoffizielle Meinung des Verfassers dieser Zeilen, dass
[§ 5 Abs. 1 UrhG][] über die Gemeinfreiheit amtlicher Werke entsprechend
angewendet werden kann.


[SKGB]: https://www.skgb.de/
[Kontakt]: https://www.skgb.de/kontakt
[XML]: https://de.wikipedia.org/wiki/Extensible_Markup_Language
[XSL]: https://de.wikipedia.org/wiki/Extensible_Stylesheet_Language
[§ 5 Abs. 1 UrhG]: http://www.gesetze-im-internet.de/urhg/__5.html


CHANGES.md
Änderungsgeschichte
===================

| MV/*Beschl.* | Satz | FinO  | StegO | HausO | ANutzO | NutzOVayu | NutzOvG | DSE   | JO    | NutzOJuB | JüngSV |
| :----------: | :--: | :---: | :---: | :---: | :----: | :-------: | :-----: | :---: | :---: | :------: | :----: |
| *1997-04-19* | ...  | ...   | ...   | ...   | ...    | ...       | ...     | ...   | ...   | ...      | *neu*  |
| 1999-10-23   | ...  | ...   | ...   | ...   | ...    | neu       | ...     | ...   | ...   | ...      | \|     |
| 2000-03-11   | ...  | neu   | neu   | neu   | ...    | \|        | ...     | ...   | ...   | neu      | \|     |
| 2001-03-17   | neu  | Änd.  | \|    | \|    | ...    | \|        | ...     | ...   | neu   | Änd.     | \|     |
| 2001-10-27   | \|   | Änd.  | \|    | Änd.  | ...    | \|        | ...     | ...   | \|    | \|       | \|     |
| *2001-11-24* | \|  | *Änd.* | \|    | \|    | ...    | \|        | ...     | ...   | \|    | \|       | \|     |
| 2002-03-09   | \|   | Änd.  | Änd.  | \|    | ...    | Änd.      | ...     | ...   | \|    | \|       | \|     |
| 2003-11-08   | \|   | \|    | Änd.  | \|    | ...    | \|        | ...     | ...   | \|    | \|       | \|     |
| 2004-03-20   | \|   | \|    | \|    | \|    | ...    | Änd.      | ...     | ...   | \|    | \|       | \|     |
| 2006-03-18   | Änd. | Änd.  | \|    | \|    | ...    | \|        | ...     | ...   | \|    | \|       | \|     |
| 2008-03-08   | \|   | \|    | \|    | Änd.  | ...    | \|        | ...     | ...   | \|    | \|       | \|     |
| *2010-03-05* | \|   | \|    | \|    | \|    | ...    | *Änd.*    | ...     | ...   | \|    | \|       | \|     |
| 2010-04-17   | \|   | Änd.  | Änd.  | \|    | ...    | \|        | ...     | ...   | Änd.  | Änd.     | \|     |
| *2010-09-04* | \|  | *Änd.* | \|    | \|    | ...    | \|        | ...     | ...   | \|    | \|       | \|     |
| 2011-05-28   | \|   | Änd.  | \|    | \|    | ...    | Änd.      | ...     | ...   | \|    | \|       | \|     |
| *2012-02-11* | \|   | \|    | \|    | \|    | ...    | *Änd.*    | ...     | ...   | \|    | \|       | \|     |
| 2012-03-17   | \|   | Änd.  | \|    | \|    | ...    | \|        | neu     | ...   | \|    | \|       | \|     |
| 2014-04-05   | \|   | Änd.  | \|    | \|    | ...    | \|        | \|      | ...   | \|    | \|       | \|     |
| 2015-05-30   | Änd. | \|    | \|    | \|    | ...    | \|        | \|      | neu   | \|    | \|       | \|     |
| 2016-04-09   | Änd. | Änd.  | \|    | \|    | ...    | \|        | \|      | \|    | \|    | \|       | \|     |
| *2018-01-26* | \|   | *Änd.*| \|    | \|    | ...    | \|        | \|      | \|    | \|    | \|       | \|     |
| 2018-03-17   | \|   | \|    | \|    | \|    | ...    | \|        | \|      | \|    | Änd.  | \|       | \|     |
| 2019-03-23   | \|   | \|    | \|    | \|    | neu    | entf.     | entf.   | \|    | \|    | \|       | \|     |

Nur die Historie der aktuellen Generation des Regelwerks ist dargestellt.
Die Satzung und Jugendordnung von 2001 lösten die Satzung (mit Jugendsatzung)
von 1980 ab. Die Steg- und Liegeplatzordnung von 2000 löste die Stegordnung
von 1994 ab.


Abkürzungen
-----------

**MV** – Mitgliederversammlung (Datum)  
_**Beschl.** – Beschluss des Vorstands oder des zuständigen Gremiums (Datum der Vorstandssitzung o. ä.)_

**Satz** – Satzung  
**FinO** – Finanzordnung  
**StegO** – Steg- und Liegeplatzordnung  
**HausO** – Bootshausordnung  
**ANutzO** – Allgemeine Vereinsboots-Nutzungsordnung  
**NutzOVayu** – Sailhorse-Nutzungsordnung  
**NutzOvG** – Lelievlet-Nutzungsordnung  
**DSE** – Datenschutzerklärung  
**JO** – Jugendordnung  
**NutzOJuB** – Jugendboot-Nutzungsordnung  
**JüngSV** – Jüngstensegel-Vorschrift

**neu** – neu verabschiedet (ersetzt u. U. ein älteres Werk)  
**Änd.** – Änderung  
**entf.** – entfällt  
**Pl.** – geplant (in Vorbereitung)


script/SKGB/Regeln.pm
package SKGB::Regeln;

use 5.014;
use utf8;

our $VERSION = 0.81;

use Cwd 'abs_path';
use File::Basename 'dirname';
#use File::Spec;
use XML::LibXML 1.70 qw();
use XML::LibXSLT;
#use Devel::StackTrace qw();
#use Data::Dumper qw();

#use Carp qw();
#use POSIX qw();


sub load {
	my ($class, $location) = @_;
	my $instance = {
		path => $location,
		doc => XML::LibXML->load_xml(location => $location),
		xslt => undef,
		versions => undef,
		html_stylesheet => (dirname abs_path $location) . '/regeln2html.xsl',
		odf_stylesheet => (dirname abs_path $0) . '/regeln2odf.xsl',
	};
	return bless $instance, $class;
}


sub path {
	my ($self) = @_;
	return $self->{path};
}


sub regeln_info {
	my ($self) = @_;
	my $context = XML::LibXML::XPathContext->new($self->{doc});
	$context->registerNs('regeln', 'http://www.skgb.de/2005/regeln');
	my $version = $context->findnodes('/regeln:regeln/@version');
	$version or return undef;
	my $info = { regeln_version => $version->string_value() };
	my $fullname = $context->findnodes('/regeln:regeln/*/regeln:name');
	$fullname and $info->{fullname} = $fullname->string_value();
	my $shortname = $context->findnodes('/regeln:regeln/*/regeln:name/@abkuerzung');
	$shortname and $info->{shortname} = $shortname->string_value();
	my $colloquial = $context->findnodes('/regeln:regeln/*/regeln:name/@umgangssprachlich');
	$colloquial and $info->{colloquial} = $colloquial->string_value();
	my $stand = $context->findnodes('//regeln:stand');
	$stand and $info->{stand} = $stand->string_value();
	my $aktuell = $context->findnodes('//regeln:aktuell/@den');
	$aktuell and $info->{aktuell} = $aktuell->string_value();
	return $info;
}


sub xslt {
	my ($self) = @_;
	$self->{xslt} ||= XML::LibXSLT->new();
	return $self->{xslt};
}


sub _transform {
	my ($self, $style_doc) = @_;
	my $stylesheet = $self->xslt->parse_stylesheet($style_doc);
	my $results = $stylesheet->transform($self->{doc});
	my $new_source = XML::LibXML->load_xml( string => $stylesheet->output_as_chars($results), load_ext_dtd => 0 );
	return $new_source;
}


sub _transform_stylesheet {
	my ($self, $style) = @_;
	$self->{doc} = $self->_transform(XML::LibXML->load_xml( location => $style, no_cdata => 1 ));
}


sub _transform_singlerule {
	my ($self, $style) = @_;
	$style or return;# $self->{doc};
#	print STDERR "---\n$style\n";
	# based on http://www.usingxml.com/Transforms/XslIdentity
	$self->{doc} = $self->_transform(XML::LibXML->load_xml( string => <<'EOT' . $style . '</xsl:transform>', no_cdata => 1 ));
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:transform version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns="http://www.skgb.de/2005/regeln" xmlns:regeln="http://www.skgb.de/2005/regeln" xmlns:html="http://www.w3.org/1999/xhtml" exclude-result-prefixes="regeln">
<xsl:output method="xml" media-type="application/xml" encoding="UTF-8" omit-xml-declaration="no" standalone="yes" indent="no"/>
<xsl:template match="/ | @* | node()">
	<xsl:copy>
		<xsl:apply-templates select="@* | node()" />
	</xsl:copy>
</xsl:template>
EOT
#	return $self->doc;
}


sub version_list {
	my ($self) = @_;
	open (FILE, '<', $self->{path}) or die ("ERROR: Could not read file '" . $self->{path} . "'");
	my %dates = ();
	while (<FILE>) {
#		while (/=['"][^'"]*(\d\d\d\d-\d\d-\d\d)[^'"]*['"][^>]*>/g) {  # bug: in case of multiple dates (e.g. in @nachgefuehrtam), only one is recognised
		next if m/^\$Id: /;
		while (m/['" ](\d\d\d\d-\d\d-\d\d)['" ]/g) {
#			print pos, "\n";
			$dates{$1} = 1;
		}
	}
	close (FILE);
	return sort {$a cmp $b} keys %dates;
}


sub show_base {
	my ($self, $baseVersion) = @_;
	if ($baseVersion ne 'head') {
		my @versionList = $self->version_list;
		my @hideVersions = ();
		foreach my $version (@versionList) {
			if ($baseVersion eq 'base' || $version gt $baseVersion) {
				push @hideVersions, $version;
			}
		}
	#	if (! @hideVersions) {
	#		$style_rule = undef;
	#	}
	#	else {
			@hideVersions = sort {$b cmp $a} @hideVersions;
	#		die "@hideVersions";
			my $stand = $baseVersion eq 'base' ? 'ursprünglichen Beschluss' : $baseVersion;
			for my $hideVersion (@hideVersions) {
				my $style_rule = <<"EOT";
<!-- last XSLT rule has highest precendence => is evaluated first -->
<xsl:template match=\"*[\@eingefuegtam and \@eingefuegtam='$hideVersion']\" />
<xsl:template match=\"*[\@gestrichenam and \@gestrichenam='$hideVersion'] | *[\@nachgefuehrtam and \@nachgefuehrtam='$hideVersion']\">
	<xsl:copy>
		<xsl:apply-templates select=\"\@eingefuegtam | \@hinweis | node()\" />
	</xsl:copy>
</xsl:template>
<xsl:template match=\"//regeln:ordnung | //regeln:satzung\">
		<xsl:copy>
			<xsl:attribute name="stand">$stand</xsl:attribute>
			<xsl:apply-templates select="@* | node()" />
		</xsl:copy>
</xsl:template>
EOT
				$self->_transform_singlerule($style_rule);
			}
	#	}
	}
}


sub hide_changes {
	my ($self) = @_;
	$self->_transform_singlerule(<<'EOT');
<!-- last XSLT rule has highest precendence => is evaluated first -->
<xsl:template match="*[@eingefuegtam] | *[@nachgefuehrtam]">
	<xsl:copy>
		<xsl:apply-templates select="@gestrichenam | node()" />
	</xsl:copy>
</xsl:template>
<xsl:template match="*[@gestrichenam]" />
EOT
}


sub show_changes {
	my ($self, $showChanges) = @_;
	$self->_transform_singlerule(<<"EOT");
<!-- last XSLT rule has highest precendence => is evaluated first -->
<xsl:template match=\"*[\@eingefuegtam and not(\@eingefuegtam='$showChanges')] | *[\@nachgefuehrtam and not(\@nachgefuehrtam='$showChanges')]\">
	<xsl:copy>
		<xsl:apply-templates select=\"\@gestrichenam | \@hinweis | node()\" />
	</xsl:copy>
</xsl:template>
<xsl:template match=\"*[\@gestrichenam and not(\@gestrichenam='$showChanges')]\" />
EOT
}


sub as_html {
	my ($self) = @_;
#	print transform_stylesheet($xml, 'regeln2html.xsl');
	$_ = join '', $self->_transform_stylesheet($self->{html_stylesheet});
	utf8::decode $_ or warn 'Unicode error';
	# leere Beitragslistenelemente entfernen (Folge der dortigen Probleme mit ins-del)
	s{<dt>:</dt><dd>[^<]*</dd>}{}g;
	return $_;
}


sub as_odf {
	my ($self) = @_;
	$_ = join '', $self->_transform_stylesheet($self->{odf_stylesheet});
	utf8::decode $_ or warn 'Unicode error';
	s{<meta:generator>}{<meta:generator>SKGB::Regeln/$VERSION }g;
	s{<meta:generator>}{<meta:generator>SKGB::Intern/$self->{skgb_intern_version} }g if $self->{skgb_intern_version};
	# BUG: in 9 (1) satzung kapselt <s> eine lit-Liste - böse, aber prinzipiell ja nicht falsch. was es bedeutet: <s> muss, falls wir es weiterverwenden wollen, den "satz" als einzige semantik beinhalten und die listen/textabsatz-logik muss anders entstehen (was wohl so war, wie's mal gedacht war).
	s{<text:p([^>]*)>\s*([^<>]+)\s*<text:list>}{<text:p$1>$2</text:p><text:list>}g;
	s{</text:list>\s*</text:p>}{</text:list>}g;
	s{</text:list>\s*([^<>]+)\s*</text:p>}{</text:list><text:p text:style-name="SKGB-Regeln-Buchstaben">$1</text:p>}g;
	# paragraph mit 1 absatz MIT <abs>
	s{<text:p text:style-name="SKGB-Regeln-Absatz-single"/>\s*<text:list-item>\s*<text:p text:style-name="SKGB-Regeln-Absatz">}{<text:p text:style-name="SKGB-Regeln-Absatz-single">}g;
	s{</text:list-item>\s*</text:list-item>}{</text:list-item>}g;
	# paragraph mit 1 absatz OHNE <abs>
	s{<text:p text:style-name="SKGB-Regeln-Absatz-single"/>\s*<text:p text:style-name="SKGB-Regeln-Absatz">}{<text:p text:style-name="SKGB-Regeln-Absatz-single">}g;
	# BUG: ist mitten im absatz eine lit-liste eingefuegt, wird diese hiermit uebersprungen und erst am ende eingefuegt. was wir bräuchten: CSS-+-selektor; einstweilige loesung: ein <text:p> je <regeln:s>
	s{</text:p>\s*<text:p text:style-name="SKGB-Regeln-Absatz">}{ }g;
	# BUG: [ins-del] es entstehen evtl. unnütze Leerzeichen
	s{ +}{ }g;
	s{<text:line-break/> (\w)}{<text:line-break/>$1}g;  # NEU
	# BUG: [ins-del] es entstehen evtl. Artefakte wie "Wort1 , Wort2" oder "01.09. 1967" <- noch keine Loesung
	# BUG: [regeln:lit] schlaegt fehl, wenn das lit noch irgendwas anderes enthaelt
	s{<text:p text:style-name="SKGB-Regeln-Buchstaben">\s*<text:p text:style-name="SKGB-Regeln-Absatz">}{<text:p text:style-name="SKGB-Regeln-Buchstaben">}g;
	s{</text:p>\s*</text:p>}{</text:p>}g;
#	s{([a-z]) , }{$1, }g;
	# aufhuebschen
	s{><text:list-item>}{>\n<text:list-item>}g;
	# BUG: geht's nach lit-Liste mit fliesstext weiter, soll kein vertikaler abstand entstehen
	s{</text:list>\s*<text:p text:style-name="SKGB-Regeln-Absatz">}{</text:list>\n<text:p text:style-name="SKGB-Regeln-Buchstaben">}g;
	return $_;
}


sub as_txt {
	my ($self) = @_;
	$_ = $self->xslt->parse_stylesheet(XML::LibXML->load_xml( string => <<'EOT', no_cdata => 1 ))->transform($self->{doc});
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<transform version="1.0" xmlns="http://www.w3.org/1999/XSL/Transform">
<output method="text" media-type="text/plain" encoding="utf-8" omit-xml-declaration="yes"/>
</transform>
EOT
	utf8::decode $_ or warn 'Unicode error';
	s{<\?[^\?]*\?>}{}g;  # remove XML declaration
	s{^\s*}{}mg;  # chomp
	s{\n(\d+)\n}{ $1\n}g;  # Beitragslistenelemente auf einzelne Zeilen zusammenführen
	s{ +}{ }g;  # BUG: [ins-del] es entstehen evtl. doppelte Leerzeichen
	return $_;
}


sub as_report {
	my ($self) = @_;
	die 'report / "Artikelgesetz" - not implemented';
}


sub as_xml {
	my ($self) = @_;
	$_ = $self->{doc};
	utf8::decode $_ or warn 'Unicode error';
	return $_;
}




# ./regeln.pl FinO.xml -l > FinO.list ; while read f ; do ./regeln.pl -c "$f" -f html FinO.xml > "FinO.$f.d.xhtml" ; done < FinO.list


1;

__END__


script/regeln.pl
#! /usr/bin/env perl

use strict;
use warnings;

our $VERSION = 0.80;

use Cwd 'abs_path';
use File::Basename 'dirname';
use lib dirname abs_path $0;
#use File::Spec;
use Getopt::Long 2.33 qw( :config posix_default gnu_getopt auto_version auto_help );
use Pod::Usage;
#use Data::Dumper;

use SKGB::Regeln;


# parse CLI parameters
#my $verbose = 0;
my $baseVersion = '';
my $showChanges = 'none';  # all | none | <date>
my $outputFormat = 'xml';  # xml | html | odf | txt | rep
my %options = (
	man => undef,
	list_mode => undef,
);
GetOptions(
#	'verbose|v+' => \$verbose,
	'man' => \$options{man},
	'changes|c=s' => \$showChanges,
	'base-version|b=s' => \$baseVersion,
	'format|f=s' => \$outputFormat,
	'list-versions|l' => \$options{list_mode},
) or pod2usage(2);
pod2usage(-exitstatus => 0, -verbose => 2) if $options{man};
#push @ARGV, 'Satzung/Satzung 2015.xml';  # DEBUG

# make sure we have everything we need
if (1 > scalar @ARGV) {
	pod2usage(-exitstatus => 3, -verbose => 0, -message => 'Missing required input file name.');
}


my $regeln = SKGB::Regeln->load( $ARGV[0] );


if ($options{list_mode}) {
	my @versionList = $regeln->version_list;
	print "base\n";
	foreach my $date (@versionList) {
		print "$date\n";
	}
	exit 0;
}


#pre1: remove changes newer than the given base version
if (! $baseVersion) {
	if ($showChanges eq 'none' || $showChanges eq 'all') {
		$baseVersion = 'head';
	}
	else {
#		foreach my $version (@versionList) {
#			if ($version eq $showChanges) {
				$baseVersion = $showChanges;
#				last;
#			}
#		}
	}
}
$regeln->show_base($baseVersion);


#pre2
if ($showChanges eq 'none') {
	$regeln->hide_changes();
}
elsif ($showChanges ne 'all') {
	$regeln->show_changes( $showChanges );
}


my $out;
if ($outputFormat eq 'html') {
	$out = $regeln->as_html;
}
elsif ($outputFormat eq 'odf') {
	$out = $regeln->as_odf;
}
elsif ($outputFormat eq 'txt') {
	$out = $regeln->as_txt;
}
elsif ($outputFormat eq 'rep') {
	$out = $regeln->as_report;
}
else {
	$out = $regeln->as_xml;
}
utf8::encode $out;
print $out;




exit 0;

__END__

=pod

=head1 NAME

regeln.pl - handle SKGB-Regeln files

=head1 SYNOPSIS

 regeln.pl -f odf Satzung.xml  > Satzung.2015.fodt
 regeln.pl -b base Satzung.xml  > Satzung.2000.xml
 regeln.pl -c all -f html Satzung.xml  > Satzung.2000-2015.xhtml
 regeln.pl -l Satzung.xml
 regeln.pl -c 2015-05-30 Satzung.xml  > Satzung.2006-2015.xml
 regeln.pl --help|--version|--man

=head1 DESCRIPTION

Reads an SKGB-Regeln file and transforms it into the specified revision and
format.

=head1 OPTIONS

=over

=item B<--base-version, -b>

<head>, <base> or a specific version. If omitted, the default is dependant upon
the value of the --show-changes option as follows:

-c none (or omitted): The base version default is <head> (the latest version).

-c all: The base version default is <head> as well (the latest version).

-c <specific version>: The base version default is same version, such that
exactly one revision step is shown in the output.

=item B<--format, -f>

Chose an output format. Default is <xml>. See FORMATS below.

=item B<--help, -?>

Display a help message and exit.

=item B<--list-versions, -l>

List all revisions mentioned in the file specified and exit (i. e., not
including the base version).

=item B<--man>

Print the manual page and exit.

=item B<--show-changes, -c>

This option causes the document's history to be highlighted. May not be
supported by all output formats. Possible values:

<all>, <none>, or a <specific version> (usually in ISO 8601 date format) to
highlight just the changes made in the specified version since the previous
version. Default is <none>.

=item B<--verbose, -v>

Verbose mode. Produces more output about what the program does. Giving this
option multiple times may produce more and more output.

=item B<--version>

Display version information and exit.

=back

=head1 FORMATS

=over

=item B<XML <xml>>

SKGB-Regeln format.

=item B<HTML <html>>

XHTML 5 output with a stylesheet showing all changes and with a user-selectable
alternative to hide the changes.

=item B<Open Document <odf>>

The Open Document template used does not implement viewing changes at this
time. Results for --show-changes values other than 'none' are undefined.

=item B<Plain Text <txt>>

Plain text removes ALL formatting and meta-data, but displays ALL character
data. Therefore it is only well-defined for --show-changes=none.

The plain text output format is not stable. It may change to e. g. markdown in
the future. Currently it's pretty much unreadable, but its completeness may
still prove useful for tracking changes.

=item B<Report <rep>>

"Artikelgesetz"

not implemented

=back

=head1 AUTHOR

Arne Johannessen

=head1 COPYRIGHT

Copyright (c) 2015-16 THAWsoftware, Arne Johannessen

=cut
